wasm-runtime/polkadot/src/codec/slicable.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Serialisation.
+
+use runtime_support::{Vec, size_of, transmute, uninitialized, slice};
```

Comment:
using free functions like this is bad style. typically one would use them like `mem::size_of` or `mem::transmute`.

Everything here (except for Vec) is available in `libcore` too, so there's no need to import it from  the `runtime_support` crate.